pkg/clients/cluster/cluster.go

```diff
@@ -774,6 +748,258 @@ func LateInitializeSpec(spec *v1beta1.GKEClusterParameters, in container.Cluster
 
 }
 
+// newAddonsConfigUpdateFn returns a function that updates the AddonsConfig of a cluster.
+func newAddonsConfigUpdateFn(in *v1beta1.AddonsConfig) UpdateFn {
+	return func(ctx context.Context, s *container.Service, name string) (*container.Operation, error) {
+		out := &container.Cluster{}
+		GenerateAddonsConfig(in, out)
+		update := &container.UpdateClusterRequest{
+			Update: &container.ClusterUpdate{
+				DesiredAddonsConfig: out.AddonsConfig,
+			},
+		}
+		return s.Projects.Locations.Clusters.Update(name, update).Context(ctx).Do()
```

Comment:
Is this returning us to both the controller and client packages needing to know how to make `*container.Service` calls?